util/project_allocation_mapper.py

```diff
@@ -99,9 +99,17 @@ def _get_user_from_portal_db(self, username):
 
     def get_all_projects(self):
         projects = {}
-        for proj in portal_proj.objects.all():
-            proj = self.portal_to_tas_proj_obj(proj, fetch_balance=False)
-            projects[proj['chargeCode']] = proj
+        project_pi_info = {}
+        for proj in portal_proj.objects.select_related('pi').all():
```

Comment:
As an alternative, what about selecting all the related things at this level?

```
portal_proj.objects.select_related('pi', 'type', 'field', 'allocations', 'allocations__requestor').all()
```

I'm not sure if that even works, specifically the `__requestor` accessor. But it may compress everything to a single SELECT w/ multiple joins.

Reply:
Not sure about "allocations" and "allocations__requestor". but since we only use "pi", adding more things and making it a more complex query will increase the exec time.